providers/providers_manager.go

```diff
@@ -45,6 +45,51 @@ type ProviderManager struct {
 	cleanupInterval time.Duration
 }
 
+type options struct {
+	cleanupInterval time.Duration
+	cache           lru.LRUCache
+}
+
+// Option is a function that sets a provider manager option.
+type Option func(*options) error
```

Comment:
you can probably get away with having this be
```
type Option func(*ProviderManager) error
```

and set the properties directly on the pm, without the intermediary struct.